tests/0151-purge-brokers.c

```diff
@@ -0,0 +1,410 @@
+                    rd_kafka_brokers_learned_ids(rk, &actual_broker_id_cnt);
+                continue_requesting_metadata = test_clock() <= abs_timeout_us;
+
+                continue_requesting_metadata &= !fetch_metadata_verify_brokers(
+                    expected_broker_ids, expected_broker_id_cnt,
+                    actual_broker_ids, actual_broker_id_cnt);
+
+                if (await_after_action_cb)
+                        continue_requesting_metadata |=
+                            await_after_action_cb(action);
+
+        } while (continue_requesting_metadata);
+
+        TEST_ASSERT(actual_broker_id_cnt == expected_broker_id_cnt,
+                    "expected %" PRIusz " brokers in cache, got %" PRIusz,
+                    expected_broker_id_cnt, actual_broker_id_cnt);
+
+        for (i = 0; i < actual_broker_id_cnt; i++) {
+                TEST_ASSERT(actual_broker_ids[i] == expected_broker_ids[i],
+                            "expected broker id[%" PRIusz
+                            "] to be "
+                            "%" PRId32 ", got %" PRId32,
+                            i, expected_broker_ids[i], actual_broker_ids[i]);
+        }
+        RD_IF_FREE(actual_broker_ids, rd_free);
+}
+
+/**
+ * @brief Test adding and removing brokers from the mock cluster.
+ *        Verify that the client is updated with the new broker list.
+ *
+ *        All \p actions are executed in sequence.
+ *
+ *        After each action, the client is expected to have the broker
+ *        ids in \p expected_broker_ids and the count to be
+ *        \p expected_brokers_cnt .
+ *        Brokers are added at the head of the list.
+ *
+ *        @param initial_cluster_size Initial number of brokers in the cluster.
+ *        @param actions Array of actions to perform. Each action is a pair
+ *                       (action,broker id). 0 to remove, 1 to add,
+ *                       2 to set down, 3 to set up.
+ *        @param expected_broker_ids Array of broker ids expected after each
+ *                                   action.
+ *        @param expected_broker_ids_cnt Number of elements in
+ *                                       \p expected_broker_ids .
+ *        @param expected_brokers_cnt Array of expected broker count after each
+ *                                    action.
+ *        @param await_verification If `rd_false`, the verification is
+ *                                  done only after last action.
+ */
+#define TEST_ACTION_REMOVE_BROKER         0
+#define TEST_ACTION_ADD_BROKER            1
+#define TEST_ACTION_SET_DOWN_BROKER       2
+#define TEST_ACTION_SET_UP_BROKER         3
+#define TEST_ACTION_SET_GROUP_COORDINATOR 4
+#define TEST_GROUP                        "topic1"
+static void
+do_test_add_remove_brokers(int32_t initial_cluster_size,
+                           int32_t actions[][2],
+                           int32_t expected_broker_ids[][5],
+                           size_t expected_broker_ids_cnt,
```

Comment:
change this to `action_size` or something like that. We are using this count for other comparison as well.

Reply:
The count must be the same, after every action there's an `expected_broker_ids`.
```C
 *        After each action, the client is expected to have the broker
 *        ids in \p expected_broker_ids and the count to be
 *        \p expected_brokers_cnt .
```